core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -2042,6 +2057,108 @@ private long copyUnordered(RecordCursor cursor, TableWriterAPI writer, RecordToR
         return rowCount;
     }
 
+    private void createMatView(final ExecutionModel model, SqlExecutionContext executionContext) throws SqlException {
+        final CreateMatViewModel createMatViewModel = (CreateMatViewModel) model;
+        final ExpressionNode name = createMatViewModel.getName();
+
+        // Fast path for CREATE MATERIALIZED VIEW IF NOT EXISTS in scenario when the view already exists
+        final int status = executionContext.getTableStatus(path, name.token);
+        if (createMatViewModel.isIgnoreIfExists() && status == TableUtils.TABLE_EXISTS) {
```

Comment:
There are no tests for the IF NOT EXISTS option.

Reply:
There is one actually:
```
CreateMatViewTest::testCreateMatViewWithExistingTableNameTest()
```

Will improve this though.